src/test/java/org/jitsi/utils/queue/PacketQueueTests.java

```diff
@@ -30,101 +30,63 @@
-        {
-            Assertions.fail("Expected that blocked thread notified immediately "
-                + "about item added to queue");
-        }
-    }
+        blockedExecutor.start();
 
-    @Test
-    public void testClosingQueueImmediatelyNotifiesAllThreadsBlockedOnGet()
-        throws Exception
-    {
-        final DummyQueue dummyQueue = new DummyQueue(10);
-        final ArrayList<CompletableFuture<DummyQueue.Dummy>>
-            dummyItems = new ArrayList<>();
-        for (int i = 0; i < ForkJoinPool.getCommonPoolParallelism(); i++)
-        {
-            dummyItems.add(CompletableFuture.supplyAsync(dummyQueue::get));
-        }
+        Thread.sleep(200);
 
-        for (CompletableFuture<DummyQueue.Dummy> dummyItem : dummyItems)
-        {
-            try
-            {
-                // This block surrounded with try/catch necessary to ensure that
-                // thread calling PacketQueue::get blocked before items is added
-                // to queue. Giving 200 ms for CompletableFuture to stuck on get.
-                dummyItem.get(200, TimeUnit.MILLISECONDS);
-                Assertions.fail("There is no items in queue, must not be here");
-            }
-            catch (TimeoutException e)
-            {
-                // no item is added during 200 ms into queue.
-            }
-        }
+        /* No tasks executed */
+        Assertions.assertEquals(tasksExecuted.get(), 0);
 
         dummyQueue.close();
 
-        for (CompletableFuture<DummyQueue.Dummy> dummyItem : dummyItems)
-        {
-            try
-            {
-                // checks that thread stuck in PacketQueue::get notified
-                // "immediately" when PacketQueue is stopped. Giving 1 ms for
-                // CompletableFuture to transit to completed state.
-                final DummyQueue.Dummy poppedItem = dummyItem.get(
-                    1, TimeUnit.MILLISECONDS);
-                Assertions.assertNull(poppedItem, "When PacketQueue is closed "
-                    + "null must be returned");
-            }
-            catch (TimeoutException e)
-            {
-                Assertions
-                    .fail("Expected that blocked thread notified immediately "
-                        + "when queue is stopped");
-            }
-        }
+        blockedExecutor.shutdown();
+
+        dummyQueue.close();
+
+        /* Still no tasks executed */
+        Assertions.assertEquals(tasksExecuted.get(), 0);
     }
 
     @Test
     public void testAddingWhenCapacityReachedRemovesOldestItem()
+        throws Exception
     {
         final int capacity = 10;
-        final DummyQueue dummyQueue = new DummyQueue(capacity);
+        final BlockedExecutor blockedExecutor
+            = new BlockedExecutor();
+        final AtomicInteger tasksExecuted = new AtomicInteger(0);
+
+        final DummyQueue dummyQueue = new DummyQueue(capacity,
+            pkt -> {
+                Assertions.assertNotEquals(0, pkt.id);
```

Comment:
nit: This was non-obvious to me, maybe add a comment or a message for the assertion?